### PR TITLE
feat: [maps-places] new module for maps-places

### PR DESCRIPTION
**Pull request by alicejli**
Updated https://github.com/googleapis/google-cloud-java/pull/9419 to match other maps modules.

Command to execute:
```
python3.9 generation/new_client/new-client.py generate \
  --api_shortname=maps-places \
  --proto-path=google/maps/places \
  --name-pretty="Places API (New)" \
  --product-docs="https://developers.google.com/maps/documentation/places/web-service/" \
  --api-description="The Places API allows developers to access a variety of search and
    retrieval endpoints for a Place." \
  --api-id=places.googleapis.com \
  --cloud-api=false \
  --requires-billing=true \
  --distribution-name="com.google.maps:google-maps-places"
```

### CODE REVIEW

File: java-maps-places/.repo-metadata.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "api_shortname": "maps-places",
+  "name_pretty": "Places API (New)",
+  "product_documentation": "https://developers.google.com/maps/documentation/places/web-service/",
+  "api_description": "The Places API allows developers to access a variety of search and\n    retrieval endpoints for a Place.",
```
Comment (meltsufin) [inline java-maps-places/.repo-metadata.json:5]: Is the newline here intentional?

File: java-maps-places/.repo-metadata.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "api_shortname": "maps-places",
+  "name_pretty": "Places API (New)",
```
Comment (meltsufin) [inline java-maps-places/.repo-metadata.json:3]: Is this the actual requested pretty name, with the `(New)`?

Comment (alicejli) [reply java-maps-places/.repo-metadata.json:3]: According to their service yaml file, yes - but I'll confirm on the bug.

Comment (alicejli) [reply java-maps-places/.repo-metadata.json:3]: Confirmed that is indeed the name and matches the name in [Pantheon](https://pantheon.corp.google.com/marketplace/product/google/places.googleapis.com?project=boq-prod-mapsplaces)

**Comment by alicejli** [comment]
Adding "do not merge" until the team confirms on b/281587773 they are okay with releasing.

**Review by burkedavison — APPROVED** [review]

**Pull request merged**